processor/elasticapmprocessor/internal/routing/data_stream_test.go

```diff
@@ -165,3 +165,110 @@ func TestEncodeErrorDataStream(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, "default", dataStreamNamespace.Str())
 }
+
+func TestRouteMetricDataPoint(t *testing.T) {
+	tests := []struct {
+		name            string
+		setupFn         func(pcommon.Map)
+		metricName      string
+		hasServiceName  bool
+		expectedDataset string
+	}{
+		{
+			name: "internal metric with service name",
+			setupFn: func(attrs pcommon.Map) {
+				// No special attributes
+			},
+			metricName:      "golang.heap.allocations.active",
+			hasServiceName:  true,
+			expectedDataset: "apm.internal",
+		},
+		{
+			name: "non-internal metric with service name",
+			setupFn: func(attrs pcommon.Map) {
+				// No special attributes
+			},
+			metricName:     "my.custom.metric",
+			hasServiceName: true,
+			// Should not set dataset, keeping resource-level
+			expectedDataset: "",
+		},
+		{
+			name: "metric without service name",
+			setupFn: func(attrs pcommon.Map) {
+				// No special attributes
+			},
+			metricName:      "my.custom.metric",
+			hasServiceName:  false,
+			expectedDataset: "apm.internal",
+		},
+		{
+			name: "service_summary metric",
+			setupFn: func(attrs pcommon.Map) {
+				attrs.PutStr("metricset.name", "service_summary")
+			},
+			metricName:      "some.metric",
+			hasServiceName:  true,
+			expectedDataset: "apm.internal",
+		},
+		{
+			name: "service_summary with interval",
+			setupFn: func(attrs pcommon.Map) {
+				attrs.PutStr("metricset.name", "service_summary")
+				attrs.PutStr("metricset.interval", "10s")
+			},
+			metricName:      "some.metric",
+			hasServiceName:  true,
+			expectedDataset: "apm.service_summary.10s",
+		},
+		{
+			name: "transaction metric with interval",
+			setupFn: func(attrs pcommon.Map) {
+				attrs.PutStr("transaction.id", "abc123")
+				attrs.PutStr("metricset.name", "transaction")
+				attrs.PutStr("metricset.interval", "1m")
+			},
+			metricName:      "transaction.duration",
+			hasServiceName:  true,
+			expectedDataset: "apm.transaction.1m",
+		},
+		{
+			name: "span metric with interval",
+			setupFn: func(attrs pcommon.Map) {
+				attrs.PutStr("span.id", "xyz789")
+				attrs.PutStr("metricset.name", "span")
+				attrs.PutStr("metricset.interval", "30s")
+			},
+			metricName:      "span.duration",
+			hasServiceName:  true,
+			expectedDataset: "apm.span.30s",
```

Comment:
is `span` correct here? not sure

Reply:
These are the existing metricsets with interval: 
- APM transaction metrics: `metrics-apm.transaction.<metricset.interval>-<namespace>`
- APM service destination metrics: `metrics-apm.service_destination.<metricset.interval>-<namespace>`
- APM service transaction metrics: `metrics-apm.service_transaction.<metricset.interval>-<namespace>`
- APM service summary metrics: `metrics-apm.service_summary.<metricset.interval>-<namespace>`

There is no `span` metricset with interval. 
Would be good to add testcases for `service_transaction` and `service_destination` metrics.